testdata/foo.go

```diff
@@ -11,3 +11,7 @@ func paramNamesCapitalized0(X, Y int) {}
 func paramNamesCapitalized1(X int) (Y, Z int) { return 0, 0 }
 
 func (PN paramNames) Capitalized2() {}
+
+func badReturn() [](func()) {
```

Comment:
You can also add more complex tests like:
```go
func f() [](func([](func()))) {
	return nil
}
```
I think current implementation will only find outermost parenthesis and will not suggest optimal `[]func([]func())` form.

If it's this way, either add a TODO or make a note in referenced issue.
We can improve on this.

Reply:
You are right, i had added TODO for that.